index.js

```diff
@@ -1,20 +1,20 @@
 'use strict';
 const net = require('net');
 
-const getPort = port => new Promise((resolve, reject) => {
+const getPort = (port, host) => new Promise((resolve, reject) => {
 	const server = net.createServer();
 
 	server.unref();
 	server.on('error', reject);
 
-	server.listen(port, () => {
+	server.listen({port, host}, () => {
 		const port = server.address().port;
 		server.close(() => {
 			resolve(port);
 		});
 	});
 });
 
-module.exports = preferredPort => preferredPort ?
-	getPort(preferredPort).catch(() => getPort(0)) :
+module.exports = (preferredPort, preferredHost) => preferredPort ?
```

Comment:
It should be possible to do `getPort('127.0.0.1')` to get a random host without having a preferred port. This makes the API a bit kludgy though, so I think we should just go with an options object.

So a user can use either:

- `getPort()`
- `getPort(32)`
- `getPort({port: 32})`
- `getPort({host: '127.0.0.1'})`
- `getPort({port: 32, host: '127.0.0.1'})`

And we keep `getPort(32)` only for compatibility reasons, but don't document it.